### PR TITLE
applepay - add paymentRequest prop

### PR DESCRIPTION
**Pull request by andyfen**
Adds `paymentRequest` config prop to pass in values required upfront, mirroring [applepaypaymentrequest](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaypaymentrequest) for supported fields, currently `requiredShippingContactFields` and `shippingContact`

gql call made to fetch merchant `countryCode`, and buyerCountry `supportedNetworks` 

moves `createOrder` into applepaySession `validateMerchant` callback to avoid 1 sec time limit between click and new ApplepaySession, then updates payment sheet mapped from order.

**depends on** 
https://github.paypal.com/Checkout-R/xobuyernodeserv/pull/2698

https://github.com/paypal/paypal-checkout-components/pull/1963

#### Integration
```
paypal.Buttons({
    style: {
        label: "pay",
    },
    paymentRequest: {
        applepay: {
            requiredShippingContactFields: [
                "postalAddress",
                "name",
                "phone",
                "email"
            ],
            shippingContact: {
                locality: "Cupertino",
                country: "United States",
                postalCode: "95014-2083",
                administrativeArea: "CA",
                emailAddress: "ravipatel@example.com",
                familyName: "Patel",
                givenName: "Ravi",
                addressLines: [
                    "1 Infinite Loop"
                ],
                countryCode: "US",
                phoneNumber: "(408) 555-5555"
            }
        },
    },
    onCancel() {
        logResponse("onCancel called", {});
    },
    createOrder(data, actions) {
        return actions.order.create(order);
    },
});

```

ship

### CODE REVIEW

**Review by mnicpt — CHANGES_REQUESTED** [review]
Has this prop been approved? Why is it needed as a prop?

**Review by wsbrunson — APPROVED** [review]

File: src/api/config.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+              clientId: $clientId
+              merchantId: $merchantId
+            ) {
+              merchantCountry,
+              supportedNetworks
+            }
+          }
+        `,
+        variables: { buyerCountry, clientId, merchantId },
+        headers:   {
```
Comment (gregjopa) [inline src/api/config.js:43]: Can we remove the unused `headers` part?
```suggestion
```

Comment (andyfen) [reply src/api/config.js]: gone

**Review by gregjopa — APPROVED** [review]

**Comment by sioked** [comment]
@andyfen can you work on increasing the coverage on this as well? I think you should be able to squash & merge as is, but I'd love to have the coverage a bit higher here.

**Comment by gregjopa** [comment]
> @andyfen can you work on increasing the coverage on this as well? I think you should be able to squash & merge as is, but I'd love to have the coverage a bit higher here.

I agree about improving coverage. Let's tackle that in a future PR. Andy is OOO today so I'm going to merge this as is.

**Pull request merged**